bundles/org.eclipse.equinox.p2.publisher.eclipse/src/org/eclipse/equinox/internal/p2/publisher/eclipse/MacOsBundleUrlType.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2024 SAP SE and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    SAP SE - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.equinox.internal.p2.publisher.eclipse;
+
+import org.eclipse.equinox.p2.publisher.eclipse.IMacOsBundleUrlType;
+
+public class MacOsBundleUrlType implements IMacOsBundleUrlType {
```

Comment:
can we use a Java Record here?

Reply:
thought about that, but I opted for consistency with the other interfaces / minimal implementations in the package

Reply:
ok

Reply:
I'm also in favor of using records. Sorry I just read this remark after assembling my other commit and suggested it there already.
In order to be consistent with other code in this plugin I suggest to use records wherever suitable.

In PDE I recently submitted a change where I even moved the record from a separate file into the only caller of the constructor, because having an (effectively) one-line in a separate file does not make sense to me.